common/sgx/qeidentity.c

```diff
@@ -50,28 +36,10 @@ oe_result_t oe_enforce_qe_identity(sgx_report_body_t* qe_report_body)
     {
         // No qe_identity info returned from the quote provider, this could be
         // because either get_qe_identity_info API was not supported or
-        // unexpected error. In both cases, check against hardcoded quoting
-        // enclave properties instead Assert that the qe report's MRSIGNER
-        // matches Intel's quoting. We will remove these hardcoded values once
-        // the libdcap_quoteprov.so was updated to support qe identity feature.
-
-        // enclave's mrsigner.
-        if (!oe_constant_time_mem_equal(
-                qe_report_body->mrsigner, g_qe_mrsigner, sizeof(g_qe_mrsigner)))
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "mrsigner mismatch");
-
-        if (qe_report_body->isvprodid != g_qe_isvprodid)
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "isvprodid mismatch");
-
-        if (qe_report_body->isvsvn < g_qeisvsvn)
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "isvsvn is out-of-date");
-
-        // Ensure that the QE is not a debug supporting enclave.
-        if (qe_report_body->attributes.flags & SGX_FLAGS_DEBUG)
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "QE has SGX_FLAGS_DEBUG set!!");
-
-        result = OE_OK;
-        goto done;
+        // unexpected error. Both cases are error conditions.
+        OE_RAISE_MSG(
+            OE_VERIFY_FAILED,
+            "unable to retrieve qe identity from DCAP quote provider");
```

Comment:
This needs to specify a NULL arg for the string to compile on Windows:
```c
OE_RAISE_MSG(OE_VERFIY_FAILED, "...", NULL);
```

See https://github.com/microsoft/openenclave/pull/1818/files for an example

Reply:
It would be great to define a new error code for this specific error case, instead of overloading OE_VERFIY_FAILED.